packages/blocks/src/paragraph-block/paragraph-service.ts

```diff
@@ -34,6 +35,24 @@ export class ParagraphBlockService extends BaseService {
         return text;
     }
   }
+
+  /**
+   * side effect when update block type
+   */
+  updateTypeEffect(model: ParagraphBlockModel, newType: string) {
+    // if change to header, move all children to parent level
+    if (newType.startsWith('h')) {
```

Comment:
You should use `isSupportChildren`

https://github.com/toeverything/blocksuite/blob/fb42cf1ce0c0ff6b4ae9fa8486fb366957e639ea/packages/blocks/src/__internal__/utils/std.ts#L6-L20

Reply:
I will update it in the next pr, where I will add test as well.